src/input/gamepad/gamepadCalibration.js

```diff
@@ -0,0 +1,232 @@
+// @flow
+/*eslint indent:0*/
+
+import {Vec2D} from "../../main/util/Vec2D";
+
+import type {ButtonInfo, StickInfo, StickCardinals, TriggerInfo, DPadInfo, GamepadInfo} from "./gamepadInfo";
+import type {Gamepad, Button} from "./gamepad";
+
```

Comment:
I've added functions to detect buttons, triggers, control sticks and d-pad. The general principle is that there would be a calibration screen, and for each piece of gamepad info to be detected, we capture two snapshots: one when nothing is being touched (which we can re-use for the rest of the time), and one when the desired input is being given.